src/main/java/ch/njol/skript/util/Time.java

```diff
@@ -24,16 +28,22 @@ public class Time implements YggdrasilSerializable, Cyclical<Integer> {
 	private final static int HOUR_ZERO = 6 * TICKS_PER_HOUR;
 	
 	private final int time;
+	private final TimeFormat timeFormat;
 
 	private static final Pattern DAY_TIME_PATTERN = Pattern.compile("(\\d?\\d)(:(\\d\\d))? ?(am|pm)", Pattern.CASE_INSENSITIVE);
 	private static final Pattern TIME_PATTERN = Pattern.compile("\\d?\\d:\\d\\d", Pattern.CASE_INSENSITIVE);
 
 	public Time() {
-		time = 0;
+		this(0);
+	}
+
+	public Time(int time) {
+		this(time, TimeFormat.TWENTY_FOUR_HOURS);
 	}
 	
-	public Time(final int time) {
+	public Time(int time, TimeFormat timeFormat) {
 		this.time = Math2.mod(time, TICKS_PER_DAY);
+		this.timeFormat = timeFormat;
 	}
```

Comment:
No checking or handling is done for AM/PM times. You can put in 16:00 pm and it's handled identically to 16:00 am, or 16:00. There should also be a way to check if 16:00 == 4:00 pm, for example.